### PR TITLE
task: update golangci-lint to latest

### PR DESCRIPTION
**Pull request by gssbzn**
## Proposed changes

Updating to latest lint version and needed fixes preventing me from committing other changes 

## Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -142,3 +142,10 @@ linters:
 #    - wastedassign # wastedassign finds wasted assignment statements. [fast: false, auto-fix: false]
 #    - wrapcheck # Checks that errors returned from external packages are wrapped [fast: false, auto-fix: false]
 #    - wsl # Whitespace Linter - Forces you to use empty lines! [fast: true, auto-fix: false]
+
+run:
+  go: '1.17'
```
Comment (gssbzn) [inline .golangci.yml:147]: ping to prevent generic warnings, we don't use generics

File: atmcfg/security.go [hunk 2]
```diff
@@ -128,18 +128,14 @@ func hmacIteration(f hashingFunc, input, salt []byte, iterationCount int) ([]byt
...
 		return nil, fmt.Errorf("error running hmacHash: %w", err)
 	}
 
+	result := make([]byte, hashSize)
 	intermediateDigest := hmacHash.Sum(nil)
-
-	for i := 0; i < len(intermediateDigest); i++ {
-		result[i] = intermediateDigest[i]
-	}
+	copy(result, intermediateDigest)
```
Comment (gssbzn) [inline atmcfg/security.go:138]: new fix

File: opsmngr/clusters_test.go [hunk 3]
```diff
@@ -74,7 +74,6 @@ func TestClusters_Get(t *testing.T) {
 	client, mux, teardown := setup()
 	defer teardown()
 
-	clusterID := "533d7d4730040be257defe88"
```
Comment (gssbzn) [inline opsmngr/clusters_test.go:77]: driveby, already defined

File: opsmngr/doc.go [hunk 4]
```diff
@@ -15,7 +15,7 @@
 /*
 Package opsmngr provides a client for using the MongoDB Ops Manager and Cloud Manager API.
 
-Usage
```
Comment (gssbzn) [inline opsmngr/doc.go:18]: linter complains since the gofmt changed in 1.19

**Review by tibulca — APPROVED** [review]
